# Projeto - Fundamentos da Descoberta de Dados

Nesse projeto trabalharemos com a base de dados de produtos de um supermercado do Chile.


In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import plotly.express as px

In [10]:
df = pd.read_csv("C:\MODULO7_PROJETOFINAL_BASE_SUPERMERCADO - MODULO7_PROJETOFINAL_BASE_SUPERMERCADO (1).csv.csv", delimiter=',')

df.head(10)

title       Marca  Preco_Normal  \
0              Pack 12 un, Leche extra proteína 1 L  Loncoleche         19788   
1   Pack 12 un, Leche chocolate receta original 1 L     Soprole         18228   
2    Pack 12 un, Leche semidescremada chocolate 1 L     Soprole         18228   
3     Pack 12 un, Leche semidescremada frutilla 1 L     Soprole         18228   
4       Pack 12 un, Leche sin lactosa chocolate 1 L  Loncoleche         17988   
5        Pack 12 un, Leche sin lactosa frutilla 1 L  Loncoleche         17988   
6  Pack 12 un, Leche saborizada light chocolate 1 L  Loncoleche         17988   
7         Pack 12 un, Leche saborizada frutilla 1 L       Colun         17388   
8         Pack 12 un, Leche saborizada vainilla 1 L       Colun         17388   
9           Pack 12 un, Leche saborizada manjar 1 L       Colun         17388   

   Preco_Desconto  Preco_Anterior  Desconto Categoria  
0               0               0         0   lacteos  
1               0               0         0   lacteos  
2               0               0         0   lacteos  
3               0               0         0   lacteos  
4               0               0         0   lacteos  
5               0               0         0   lacteos  
6               0               0         0   lacteos  
7               0               0         0   lacteos  
8               0               0         0   lacteos  
9               0               0         0   lacteos

Os campos do nosso dataframe são:

**Title:** Nome do produto.


**Marca:** A marca do produto.


**Preco_Normal:** O preço em que o produto costuma ser vendido quando não há desconto.


**Preco_Desconto:** O preço vendido após o desconto ser aplicado.


**Preco_Anterior:** Preço em que era comercializado o produto antes do desconto aplicado.


**Desconto:** Total de desconto aplicado.






As colunas que aparecem com valores 0 são para os produtos onde não tivemos descontos aplicados.


As categorias estão em espanhol!

# 1 - Traga a média e a mediana dos preços - coluna Preco_Normal - por categoria de produto.
# Identifique as categorias que parecem ter um valor de média abaixo ou acima da mediana.

In [20]:
#Seu código aqui para a média
df.groupby('Categoria')['Preco_Normal'].mean().reset_index().sort_values(by='Preco_Normal', ascending=False)

Categoria  Preco_Normal
1          comidas-preparadas   3095.043478
5                     lacteos   2385.219239
2                  congelados   2108.042553
0  belleza-y-cuidado-personal   1783.556485
3                      frutas   1724.473684
6                    verduras   1343.296875
4        instantaneos-y-sopas    765.491228

In [21]:
#Seu código aqui para a mediana
df.groupby('Categoria')['Preco_Normal'].median().reset_index().sort_values(by='Preco_Normal', ascending=False)

Categoria  Preco_Normal
1          comidas-preparadas        3290.0
0  belleza-y-cuidado-personal        1569.0
2                  congelados        1519.0
3                      frutas        1195.0
6                    verduras        1180.0
5                     lacteos         989.0
4        instantaneos-y-sopas         439.0

Lacteos, Congelados, Frutas e instantâneos e sopas 

# 2 - Traga o desvio padrão por categoria de produto.
# Qual o comportamento da média e mediana nas categorias com maior desvio?

In [23]:

desvio_padrao_por_categoria = df.groupby('Categoria')['Preco_Normal'].std().reset_index()
desvio_padrao_por_categoria

Categoria  Preco_Normal
0  belleza-y-cuidado-personal   2210.041719
1          comidas-preparadas   2019.911428
2                  congelados   2111.539896
3                      frutas   1639.151114
4        instantaneos-y-sopas   1170.232869
5                     lacteos   3925.816164
6                    verduras   1012.699625

# 3 - Plot um boxplot da distribuição do Preco_Normal para a categoria que você identificou que tem o maior desvio padrão. Como é a distribuição desses dados segundo o boxplot? Você identifica muitos outliers?

In [27]:

# Filtra os dados da categoria 'Lacteos' e cria o boxplot
fig = px.box(df.loc[df['Categoria'] == 'lacteos'], 
             y='Preco_Normal', 
             title='Boxplot da Categoria Lácteos')
# Exibir o Gráfico
fig.show()

Após o uso do Gráfico Iterativo de Box Plot, utilizando a tabela de categoria, sendo filtrando pela categoria que tivemos mais desvio padrão "Lacteos", tivemos uma análise que nossa mediana é de 989, nosso Q1 é equivalente a 269, Q3 é de 2619, nosso limite inferior equivale a 0, e, limite superior chega até aproximadamente 5934, valores acima do nosso limite superior são considerado outiliers, ou seja, valores extremos, obeservamos que temos diversos outiliers chegando até 20.000, A diferença entre o terceiro quartil e os outliers sugere que alguns produtos têm preços bem superiores aos da maioria, o que pode ser devido a produtos premium ou de tamanho maior.

# 4 - Plote um gráfico de barras onde temos a média de descontos por categoria.

In [42]:
# Calculando a média de desconto por categoria
Mdesconto_categoria = df.groupby('Categoria')['Desconto'].mean().reset_index().sort_values(by='Desconto', ascending=False)

# Criando o gráfico interativo
fig = px.bar(Mdesconto_categoria, 
              x='Categoria', 
              y='Desconto', 
              title='Média de Desconto por Categoria',
              labels={'Desconto': 'Desconto', 'Categoria': 'Categoria'},
              color='Desconto', 
              color_continuous_scale=px.colors.sequential.Blues)

# Ajustando o layout
fig.update_layout(xaxis_title='Categoria', 
                  yaxis_title='Desconto', 
                  xaxis_tickangle=-45, 
                  template='plotly_white')

# Exibir o gráfico
fig.show()

# 5 - Plote um gráfico de mapa interativo agrupando os dados por categoria, marca e trazendo a média de desconto.

In [50]:
# Agrupando os dados por Categoria e Marca e calculando a média de Desconto
Categoria_por_Marca = df.groupby(['Categoria', 'Marca'])['Desconto'].mean().reset_index()

# Removendo linhas onde Desconto é zero ou nulo
Categoria_por_Marca = Categoria_por_Marca[Categoria_por_Marca['Desconto'] > 0]

# Criando o gráfico Treemap
fig = px.treemap(Categoria_por_Marca, 
                 path=['Categoria', 'Marca'],  # Caminho hierárquico
                 values='Desconto',  # Valores para o tamanho dos retângulos
                 title='Média de Desconto por Categoria e Marca', # Definindo o Titulo
                 color='Desconto',  # Cor baseada na média de desconto
                 color_continuous_scale=px.colors.sequential.Viridis)  # Escala de cores

# Exibir o gráfico
fig.show()
